Testscripts/Windows/PERF-LKP.ps1

```diff
@@ -0,0 +1,248 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the Apache License.
+
+param(
+    [object] $AllVmData,
+    [object] $CurrentTestData,
+    [object] $TestProvider,
+    [object] $TestParams
+)
+
+$TEST_SCRIPT = "perf_lkp.sh"
+$LKP_RESULTS = "lkp-results.log"
+$LKP_OUPUT = "lkp-output.log"
+$LKP_LOG_FILE = "lkp_log.tar"
+$LKP_LOG_DIR = "lkp_log"
+$VERTICAL_FORMAT_TESTS = @("oltp", "redis", "perf-bench-numa-mem", "apachebench", "pmbench", "aim9")
+$MICROBENCHMARK_TEST_NAMES = @("perf-bench-sched-pipe", "schbench", "oltp", "redis", "perf-bench-numa-mem", "apachebench", "pmbench")
+
+function Get-BenchmarkResult ($testName, $jsonResult) {
+    switch ($testName) {
+        "hackbench" {
+            $resultTitle = "{0,-30} {1,-30}" -f "throughput", "workload"
+            $result = "{0,-30} {1,-30}" -f ${jsonResult}.'hackbench.throughput'[0], ${jsonResult}.'hackbench.workload'[0]
+        }
+        "perf-bench-sched-pipe" {
+            $resultTitle = "ops_per_sec"
+            $result = ${jsonResult}.'perf-bench-sched-pipe.ops_per_sec'[0]
+        }
+        "schbench" {
+            $resultTitle = "latency_max_us"
+            $result = ${jsonResult}.'schbench.latency_max_us'[0]
+        }
+        "pgbench" {
+            $resultTitle = "{0,-30} {1,-30} {2,30} {3,30}" -f "transactions", "latency_ms", "tps", "tps_exclude_connect"
+            $result = "{0,-30} {1,-30} {2,30} {3,30}" -f ${jsonResult}.'pgbench.transactions'[0], ${jsonResult}.'pgbench.latency_ms'[0], `
+            ${jsonResult}.'pgbench.tps'[0], ${jsonResult}.'pgbench.tps_exclude_connect'[0]
+        }
+        "unixbench" {
+            $resultTitle = "{0,-30} {1,-30}" -f "score", "workload"
+            $result = "{0,-30} {1,-30}" -f ${jsonResult}.'unixbench.score'[0], ${jsonResult}.'unixbench.workload'[0]
+        }
+        default {
+            Write-LogErr "unknow test name of ${testName}"
+            return $null,$null
+        }
+    }
+    return $resultTitle, $result
+}
+
+function Get-VerticalFormat-TestResult ($currentTestResult, $currentTestData, $testName) {
+
+    try {
+
+        $LogPath = Join-Path $LogDir\$LKP_LOG_DIR $testName
+        if (-not (Test-Path -Path $LogPath)) {
+            Write-LogErr "EXCEPTION : $LKP_LOG_DIR doesn't exist"
+            return $resultAborted
+        }
+
+        $items = Get-ChildItem $LogPath
+        $items | ForEach-Object {
+            $resultPath = Join-Path $LogPath $_
+            $resultFile = Join-Path $resultPath "${testName}.json"
+            if (-not (Test-Path -Path $resultFile)) {
+                Write-LogErr "EXCEPTION : $resultFile doesn't exist"
```

Comment:
throw statement?

Reply:
I don't want to change to throw statement for confused error log.

Reply:
This branch don't return abort. So don't use throw.